display_list/skia/dl_sk_canvas.cc

```diff
@@ -23,11 +23,21 @@ constexpr float kInvertColorMatrix[20] = {
 };
 // clang-format on
 
-static SkPaint ToSk(const DlPaint& paint, bool force_stroke = false) {
+SkPaint ToSk(const DlPaint& paint, bool force_stroke) {
   SkPaint sk_paint;
 
   sk_paint.setAntiAlias(paint.isAntiAlias());
-  sk_paint.setDither(paint.isDither());
+
+  // On the Impeller backend, we will only support dithering of *gradients*,
+  // and it will be enabled by default (without the option to disable it). Until
+  // Skia support is completely removed, we only want to respect the dither flag
+  // for gradients (otherwise it will also apply to, for example, images, which
+  // is not supported in Impeller).
+  //
+  // See https://github.com/flutter/flutter/issues/112498.
+  if (paint.isDitherHintForSkBackend()) {
```

Comment:
A simpler and more direct solution would be to check ColorSource for null down below - which will skip both the call to ToSk and the need to deal with dithering for the very common case where it is null. Down below:

```
  auto color_source = paint.getColorSourcePtr();
  if (color_source) {
    sk_paint.setDither(color_source->isDither...());
    sk_paint.setShader(ToSk(color_source));
  }
```

Reply:
Nice suggestion, done!